CMakeLists.txt

```diff
@@ -94,7 +94,7 @@ elseif (MLX_BUILD_METAL)
   FetchContent_Declare(
     metal_cpp
     URL ${METAL_CPP_URL}
-    PATCH_COMMAND patch -N -i ${METAL_CPP_PATCH} || true
+    PATCH_COMMAND /usr/bin/patch -N -i ${METAL_CPP_PATCH} || true
```

Comment:
Why this change? Was it not working without it?

Reply:
> Why this change? Was it not working without it?

As discussed in this https://github.com/ml-explore/mlx/issues/1045 issue, it does not work without adding this.

Reply:
@NripeshN your default patch wasn't working. Could you do `patch -v` and tell me the output? I tried with both the macOS default patch and gpatch and they both work fine. So I'm wondering which version of patch you have that didn't work.

Reply:
```(venv) nripeshniketan@Nripeshs-MBP ~/D/p/mlx (clip_grad_norm) [0|2]> patch -v
GNU patch 2.7.6
Copyright (C) 2003, 2009-2012 Free Software Foundation, Inc.
Copyright (C) 1988 Larry Wall

License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Larry Wall and Paul Eggert```